src/decoder/mod.rs

```diff
@@ -923,6 +956,38 @@ impl<R: Read + Seek> Decoder<R> {
         Ok(())
     }
 
+    pub fn read_jpeg(&mut self, read_strip_wise: bool) -> TiffResult<DecodingResult> {
+
+        let offsets: Vec<u32>;
+        let bytes: Vec<u32>;
+        let jpeg_tables: Vec<u8> = self.get_tag_u8_vec(Tag::JPEGTables)?;
+
+        if read_strip_wise {
+            offsets = self.get_tag_u32_vec(Tag::StripOffsets)?;
+            bytes = self.get_tag_u32_vec(Tag::StripByteCounts)?;
+        } else {
+            offsets = self.get_tag_u32_vec(Tag::TileOffsets)?;
+            bytes = self.get_tag_u32_vec(Tag::TileByteCounts)?;
+        }
+
+        let mut res_img = Vec::with_capacity(offsets[0] as usize);
+
+        for (idx, offset) in offsets.iter().enumerate() {
+            self.goto_offset(*offset)?;
+            let jpeg_reader = JpegReader::new(&mut self.reader, bytes[idx], &jpeg_tables)?;
+            let mut decoder = jpeg::Decoder::new(jpeg_reader);
+
+            match decoder.decode() {
+                Err(e) => {
+                    panic!("Error when reading jpeg tiles: {}", e);
```

Comment:
Never panic on an error from input. Instead, match the underlying [`Error`](https://docs.rs/jpeg-decoder/0.1.20/jpeg_decoder/enum.Error.html) and return the io error, or wrap into a new variant of `TiffFormatError`/`UnsupportedError` as appropriate.